dev-tools/oxlog/src/lib.rs

```diff
@@ -0,0 +1,442 @@
+// service.
+//
+// If parsing fails for some reason, return `None`.
+pub fn oxide_smf_service_name_from_log_file_name(
+    filename: &str,
+) -> Option<&str> {
+    let Some((prefix, _suffix)) = filename.split_once(':') else {
+        // No ':' found
+        return None;
+    };
+
+    for ox_prefix in OX_SMF_PREFIXES {
+        if let Some(svc_name) = prefix.strip_prefix(ox_prefix) {
+            return Some(svc_name);
+        }
+    }
+
+    None
+}
+
+// Given a directory, find all oxide specific SMF service logs and return them
+// mapped to their inferred service name.
+fn load_svc_logs(dir: Utf8PathBuf, logs: &mut BTreeMap<ServiceName, SvcLogs>) {
+    let Ok(entries) = read_dir(dir.as_path()) else {
+        return;
+    };
+    for entry in entries {
+        let Ok(entry) = entry else {
+            continue;
+        };
+        let filename = entry.file_name();
+        let Some(filename) = filename.to_str() else {
+            continue;
+        };
+
+        // Is this a log file we care about?
+        if is_oxide_smf_log_file(filename) {
+            let mut path = dir.clone();
+            path.push(filename);
+            let mut logfile = LogFile::new(path);
+
+            let Some(svc_name) =
+                oxide_smf_service_name_from_log_file_name(filename)
+            else {
+                continue;
+            };
+
+            if let Ok(metadata) = entry.metadata() {
+                if metadata.len() == 0 {
+                    // skip 0 size files
+                    continue;
+                }
+                logfile.size = Some(metadata.len());
+                if let Ok(modified) = metadata.modified() {
+                    logfile.modified = Some(modified.into());
+                }
+            }
+
+            let is_current = filename.ends_with(".log");
+
+            let svc_logs =
+                logs.entry(svc_name.to_string()).or_insert(SvcLogs::default());
+
+            if is_current {
+                svc_logs.current = Some(logfile.clone());
+            } else {
+                svc_logs.archived.push(logfile.clone());
+            }
+        }
+    }
+}
+
+// Load any logs in non-standard paths. We grab all logs in `dir` and
+// don't filter based on filename prefix as in `load_svc_logs`.
+fn load_extra_logs(
+    dir: Utf8PathBuf,
+    svc_name: &str,
+    logs: &mut BTreeMap<ServiceName, SvcLogs>,
+) {
+    let Ok(entries) = read_dir(dir.as_path()) else {
+        return;
+    };
+
+    let svc_logs =
+        logs.entry(svc_name.to_string()).or_insert(SvcLogs::default());
+
+    for entry in entries {
+        let Ok(entry) = entry else {
+            continue;
+        };
+        let filename = entry.file_name();
+        let Some(filename) = filename.to_str() else {
+            continue;
+        };
+        let mut path = dir.clone();
+        path.push(filename);
+        let mut logfile = LogFile::new(path);
+        if let Ok(metadata) = entry.metadata() {
```

Comment:
Managing metadata can be deduped here and in `load_svc_logs` by adding a method to `LogFile`. We *may* want to add a filter for metadata as well so that we don't bother collecting it if we aren't asked for it.

Reply:
When I was writing this code I realized that filtering on metadata would prevent us from checking the size or we'd end up just not reading the modified date. I don't think this is important right now, so skipped it.